src/nvim/mbyte.c

```diff
@@ -1328,28 +1239,12 @@ int utf_fold(int a)
  */
 int utf_toupper(int a)
 {
-  /* If 'casemap' contains "keepascii" use ASCII style toupper(). */
-  if (a < 128 && (cmp_flags & CMP_KEEPASCII))
-    return TOUPPER_ASC(a);
-
-#if defined(__STDC_ISO_10646__)
-  /* If towupper() is available and handles Unicode, use it. */
-  if (!(cmp_flags & CMP_INTERNAL))
-    return towupper(a);
-#endif
-
-  /* For characters below 128 use locale sensitive toupper(). */
-  if (a < 128)
-    return TOUPPER_LOC(a);
-
-  /* For any other characters use the above mapping table. */
-  return utf_convert(a, toUpper, ARRAY_SIZE(toUpper));
+  return utf8proc_toupper(a);
```

Comment:
We need to check that this is locale-sensitive in the same conditions. Likely we need to keep the checks for `cmp_flags` for this.